### PR TITLE
[3006.x] Build Mac and Windows packages without an existing onedir as well

### PR DESCRIPTION
**Pull request by MKLeb**
### What does this PR do?
See title

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by s0undt3ch — APPROVED** [review]

File: tools/pkg/build.py [hunk 1]
```diff
@@ -185,15 +184,30 @@ def macos(
...
+        build_root = checkout / "pkg" / "macos" / "build" / "opt"
+        build_root.mkdir(parents=True, exist_ok=True)
+        ctx.info(f"Extracting the onedir artifact to {build_root}")
+        with tarfile.open(str(onedir_artifact)) as tarball:
+            with ctx.chdir(onedir_artifact.parent):
+                tarball.extractall(path=build_root)
+    else:
+        ctx.info("Building package without an existing onedir")
+
+    if not onedir:
```
Comment (s0undt3ch) [inline tools/pkg/build.py:201]:
```suggestion
    else:
        ctx.info("Building package without an existing onedir")
```

**Review by s0undt3ch — APPROVED** [review]

File: tools/pkg/build.py (hunk 1 above)
Comment (s0undt3ch) [reply tools/pkg/build.py:201]: This is a nitpick and doesn't need to change to get this merged.
Approved

File: doc/topics/packaging/index.rst [hunk 2]
```diff
@@ -113,26 +119,70 @@ How to build deb packages
...
-       tools pkg build deb --relenv-version <relenv-version> --python-version <python-version> --arch <arch>
+       cd <path-to-salt-repo>
+
+#. Install the dependencies:
+
+    .. code-block:: bash
+
+       pip install -r requirements/static/ci/py{python_version}/tools.txt
+
+#. Build the MacOS package:
```
Comment (twangboy) [inline doc/topics/packaging/index.rst:178]: Build the Windows package

File: pkg/windows/build.ps1 [hunk 3]
```diff
@@ -167,7 +167,7 @@ if ( ! $SkipInstall ) {
       Architecture = $Architecture
   }
   if ( $Build ) {
-      $KeywordArguments["Build"] = $true
+      $KeywordArguments["Build"] = $false
```
Comment (twangboy) [inline pkg/windows/build.ps1:170]: I think this still needs to be `$true`

**Pull request merged**